### DISCUSSION TITLE
RFC 2: SIP-010 Fungible Token as a DAO Governance Standard

### DISCUSSION THREAD

**Discussion by JMKJR** [syvita/daoos#30] [Ideas]
# RFC 2: SIP-010 Fungible Token as a DAO Governance Standard

## Abstract
**Decentralized Autonomous Organization (DAO)** - A group of individuals on the internet that come together to trustlessly manage a Treasury and proposal objectives.

I believe the only way to have a DAO scale and avoid the pitfalls of roles, trust, and apathy is to have an SIP-010 governance token embedded at the foundation of our DAO standard. Having a token allows us to relax many of the problem constraints around “trust” that drove us towards these complex solutions in the first place.

## Benefits
* Members have skin in the game in the form of token equity. Members have economic incentive to be good stewards of the treasury and be intelligent about project proposal initiatives from a capital/labor allocation standpoint. They will collectively pursue objectives that maximize the long term value of their token equity. In this sense, complete strangers are bonded by mutual ownership incentive. This avoids [tragedy of the commons](https://en.wikipedia.org/wiki/Tragedy_of_the_commons).
* This model is a web3 improved take on an already provenly useful model - shares of corporations. Think of a highly liquid, trustless, corporation with no bosses, leaders, gatekeepers, or wage servitude. Anyone can be an owner that owns a token and you get out what you put in. No interviews, resumes, degrees, or discrimination. No public bailouts or socialized losses accrued to those outside of the DAO that encourage DAO excessive risk taking. Join and leave as you please with proof of token ownership. Do you see growth potential for a DAO? Buy their tokens, propose your suggested measures, help them grow! Tokens are a means to effective real-world action.
* This solution is already known to have product-market fit. Case in point - the Ethereum network.
* DAOs can decide on their own fair token distribution models.
* Tokens can be airdropped on users of DAO products to help bootstrap their apps. Imagine if Facebook and Instagram gave you shares of their company for using and helping grow their products and networks. Legacy (non-web3) corporations simply cannot do this!
* There is an incentive to create proposals for good ideas. Token holders are entitled to cash flows as well as future growth. This ignites the flames of governance participation in a manner not possible with members with no skin in the game.
* When there is entrepreneurial upside there is less of a reliance on grant money/donations. The Stacks ecosystem will move and grow at a faster pace.
* Tokens will attract users of other blockchains to the Stacks ecosystem.
* Simplicity is best. By rigidly defining governance around a token, we can put to rest the infinite other governance possibilities to account for via plugins and frontend UI/UX. We can still maintain the plugin concept around "proposals" but there are less degrees of freedom to worry about elsewhere now that governance possibilities and flows can be more narrowly defined.

## Risks
There are going to be risks with whatever governance model we propose. We should choose the model with the most benefits commensurate with the risks. Here I do not consider risks that would equally affect other governance models (comment thread moderation, spam etc.); these are little details that are not specific to the scope of this RFC.

* Hostile takeover - Someone can buy the needed majority of tokens for a DAO to totally control all decision making. This is not unlike the corporate world and has its debated benefits and risks. I say we accept this risk and not try to philosophically pontificate over whether this is or is not a desirable outcome. Sometime in the future, a mechanism could be provided for DAOs to ratify a constitution of sorts that establish republic style protective measures to protect from tyranny of majority. i.e certain contract operations would require a "super majority" of token approval. But I wish to avoid this type of complexity at this stage.
* There **will be** nefarious DAOs that pursue short-term token-holder value appreciation at the expense of long-term token value appreciation. An example: A DAO runs a pump and dump scheme to generate hype around projects that will never exist to dump tokens on gullible bagholders. This happens frequently on Ethereum. My take on this is - play stupid games, win stupid prizes. Investors should know the risks of what they're getting into.

> "**I would rather be exposed to the inconveniences attending too much liberty than those attending too small a degree of it.**" - Thomas Jefferson

## Conclusion
This was the initial direction I wanted to go with daoOS but I didn’t get the sense anyone else was for it. However, I feel we’ve been bumping into all of the challenges that a governance token would simply solve so maybe now that we’re looking at this again, we see the merits? What are fungible tokens for, if not for this (and stablecoins :) )? I think we’re missing a BIG opportunity here if we’re getting the chance to build the Stacks DAO standard and we don’t incorporate models/standards of what is already working on other blockchains. I apologize that what is proposed is a total rework of our project thus far but better now in the planning stage than 2 months into development.

## Actionable Items
I propose:
1. We discuss, and if there is semblance of agreement we put to a vote whether daoOS should integrate the fungible token standard.
2. If the vote passes, we get into Frontend UI/UX design and flow. We just need a minimum of the pages we want and what's going to be on the pages, the design aesthetic can be worked on in parallel with the development of the application itself. This could literally be a simple wireframe of some figma screens and even just text boxes of what needs to happen on the page. But we want to definitely know what each page is for, so we don't over engineer our architecture or backend.
3. With an understanding of the application flow and what is on each page, we can easily design an architecture (diagrams and mappings of smart contracts and what they do) to model our backend.
4. Smart contract engineers can now get to work on the backend architecture and frontend engineers can work in tandem with designers.
5. Are we going to write something like this https://github.com/stacksgov/sips/pull/23 ? If so at what stage?

Summary of actionable project stages:
Approve a governance methodology -> minimal frontend wireframe describing what screens do per updated user stories -> frontend engineers can now get to work or wait for designers to do their thing and backend architecture can now be developed. -> Backend coders can now start -> Tie everything together -> MVP done.

@syvita/daoos

**Comment by JMKJR** [comment]
This RFC presumes pre-existing knowledge pertaining to the challenges of other DAO governance approaches discussed on discord...

> Mixing economic activity (Treasury decisions, project proposals, allocation of labor/capital) with a simple democratic majority model (all equal voting members) presents a number of obvious issues:
> * Not everyone has equal skin in the game and therefore not everyone should have equal say in these policies. We are talking about real money and labor on the line here. It's simply impossible to converge on the appropriate economic course of action for a group of internet randos without the ability to share in the rewards and risk. If we relax this problem by only having DAO's become a small group of trusted members who are known to have equal skin in the game and should get equal say, then why are we even making a DAO at all? We've already discussed at length the issues and complexity with roles etc.
> * We want people to actually TRUSTLESSLY USE our product at scale. To have many DAOs that scale to big numbers, we need a way of ensuring that new members will be incentivized to vote for what they think is the most optimal outcome of a proposal as if it was their own money! An imperative constraint is that these members can join/leave at will and act in the best interests of the DAO without some authority micromanaging. i.e we need to avoid tragedy of the commons https://en.wikipedia.org/wiki/Tragedy_of_the_commons

> No small group of trust, no admins, no roles, ability to scale, have aligned interests commensurate with skin in the game.

**Comment by 314159265359879** [comment]
Agreed

**Reply by Zk2u** [reply]
this has my full support

**Comment by cryptopanter** [comment]
**I sort of disagree!**
If understood well, **you are suggesting sth like Decentralized ICO/IEO, right?**... You mean we can provide the option of token economy on the platform so it can be deployed by different teams to raise fund through their community (or any interested party) by offering them their token in return, am i right?...
If so, then the solution already exist in the crypto space as ICO/IEOs and we are only making it a native platform on Stacks!!...I feel **It's kinda inventing wheel again!!** ...Moreover, **I intrinsically disagree with ICOs or IEOs** since most of them ends in scam and doesn't lead to any tangible product later on. And that can definitely destroy the reputation of DAO as the underlying of this consequence! To be more clear, assuming successful launch of such DAO, there will exist a lot of generated tokens for various projects with many people invested in them and waiting for the final product! Since most of startups and ideas will not end up in any commercial product then the only result would be DAO reputation sabotage!

Therefore i strongly disagree with launching any ICO-shaped platform for investing in Stacks projects. **In my opinion, the best and less-risky solution is to launch sth as a crowdfunding platform** but in decentralized way. So that said, the utility token will be still STX and DAO put the liability of attracting investors on project teams themselves (They can whether promise token distribution in return or not).

**Reply by JMKJR** [reply]
Hey thanks for comments and good questions!

> you are suggesting sth like Decentralized ICO/IEO, right?

No. The objective of what is proposed is to couple treasury decisions and project initiatives with a governance token to solve a tragedy of the commons situation in which resources (labor/treasury $) are depleted by DAO members who have no skin in the game and can't be trusted because they are strangers. I'm **not** proposing an ICO platform. **There will be no feature to integrate payments to fund raise a project through selling tokens.** I do however, envision the ability to mint a governance token from within the platform. The token would be distributed differently based on the DAO's objectives and agreed upon fair token distribution model. **The tokens can be bought/exchanged off of the platform on a decentralized exchange (DEX)**. **This removes us from ICO territory.** i.e we do not provide people the market making platform to sell a bunch of tokens for an ICO. If people want to buy/sell the tokens amongst themselves they have to do so elsewhere. **Again, this is not a fund-raising platform, it is a governance platform.**

**Reply by JMKJR** [reply]
Tokens created are not sold to members or investors. They are "distributed" to members who participate in DAO activities per a DAO's fair token distribution model. But we can't stop people from ascribing value to them (because they do have value) and trading them on a secondary DEX market. (This is a good thing imo).

**Reply by cryptopanter** [reply]
Ok now i get... You mean we can make a unique token for governance of DAO so anyone is interested in participating in voting or creating proposal they need to spend that special token.. We can have it and I am currently neutral to that... I categorize your concern in two issues. 1st the concern of waste of money from the voters and 2nd concern of those creating proposals... I think for voters we cannot impose any workframe for them to follow since it's agianst the principle of democratic voting. That said, voters are responsible for their votes and decisions and are assumed mature enough to do that.. For the sake of proposal creators and project teams we can impose some soft moderation like splitting the treasury fund into several milestones so that the fund will be injected stage by stage regarding voters community monitoring.
Therfore, let's see that as two available solution:
1- Stx being to the native token for DAO (There might be some fees for devs or not)
2- we have a DAO token that those want to participate in the platform, they should buy and spend it. This will impose a token economy and play role of an incentive for those contributing to DAO project and develop it in future

**Reply by falkonprods** [reply]
The spec supports STX for now, we will eventually allow wrapped BTC, Arkadiko and SIP-010s. I think we also need to think very hard on governance models outside of proposals and voting after POC. If we can support the governance model of Stacks foundation, I have met with people there that  would be open to using this tool for governing Stacks. Which would b e awesome, of course.

**Comment by falkonprods** [comment]
I also disagree. for now.

I  do agree that eventually, this should be an option. My reasons: (a) this will be a major distraction of coding time for POC - and this is not along the lines of what the grant agreed for us to build, and (b) I have mixed feelings about coin-based governance, I  think oftentimes this is a scam way for DAOs to raise money, (c) the voting mechanism removes some of the issues that this is supposed to solve by empowering voting members to vote with  currency - this removes the issue of equal power without equal funding as voting members do not have to equally distribute the coin they can stake, (d) I don't see how having a NEW  COIN improves this  other than the  coin  raise, in  fact I  would argue using existing coins will be more stable (esp if it's Arkadiko or another stable SIP 010).

**Reply by JMKJR** [reply]
Thanks:
(a) Yeah good point, we have stakeholders to service so it could be too late to take this kind of change of direction if this product is explicitly a grant fulfillment for their requested features.

(b) You're right, it can be. But it can also be an optimal model for legitimate DAO's (see Badger DAO etc. on ETH). The surest way to have no scams is to give no one any freedom to do anything and have a dull Blockchain ecosystem that will never amount to anything (not saying this is your suggestion, just making a point here). You can have a novel decentralized consensus protocol, a better smart contract language, etc. but ultimately you need people to build an ecosystem of things people actually want. Ethereum is wildly successful because of the degrees of freedom it provides. People want governance tokens; this is not an opinion but a fact demonstrated by Ethereum.

(c) Not sure I follow what you're saying here, is this the milestone funding? If not, could you link me to where I can read more on this? If it is milestone funding, I understand that staker interests are represented with equivalent voting power to the amount they are staking. However, if members are strangers, why would anyone be incentivized to stake anything towards a shared objective where the reward of project completion is not shared according to how much one has staked? i.e what's in it for person staking the most? Sounds like charity unless i'm misunderstanding something. Strangers won't be charitable. A governance token fixes this since those who have the most of the token will accrue the most equity value from a successful project result. Entirely possible I'm totally missing something here, but I am just making sure trustless strangers are going to want to use what we make otherwise we're wasting our time (imo).

(d) Mostly explained in my response to (c). Also, you need a token to represent fractional ownership of the treasury, not just voting. Consider a scenario where the treasury needs to be liquidated - how should that be handled without shares (tokens)? A democratic vote? Again, we are talking about trustless strangers here, no? It's hard for me to imagine how a democratic vote could adequately function in this and most proposal scenarios.

**Reply by falkonprods** [reply]
I am not sure exactly what the question is but maybe walking through a couple  scenarios  might help:

Scenario 1: Freehold DAO gets proposal to create a video about why Clarity is the optimal framework for developing smart contracts. The Voting members of Freehold (assigned by Freehold manamagement) manage all of the Freehold DAO funds. Their JOB IS TO EXECUTE ON  TASKS. So they vote on this, and eventually they make it to the  100 STX that was bid to build this out. 

Scenario 2: a DAO  that  came together to create an open source app - one of the members put together a proposal to do wireframes of the proposed app. A  member who is a designer proposes creating the designs using milestones for each iteration: M1 flow, M2 low fi, M3 hi fi, M4 clickable prototype. All members of this DAO who have contributed funds are voting members (in this DAO that is the rule) and can vote what they have contributed. It is in the interest of the DAO to build these designs, so they vote on this. It doesn't get funded but the designer resubmits a scaled down version and that one does.

The purpose of a DAO is  to come together to GET  SOMETHING DONE, so this mechanism supports that and this isn't "charity", this  is  just a mechanism for deciding how to spend DAO funds.

Scenario 3:  Liquidation could happen by creating a proposal equal to the amount of the full treasury with a single milestone. In order for this to be funded, the entire treasury  would have to be staked - meaning this could only work if all of the voting  members allocate their shares to this proposal, and all approve the milestone release. Fractional withdrawals could be  made, but only in connection with the amount of treasury allotted to the voting members who  approve it.

I don't see why there are any selfish  concerns here. Honestly if someone wants to contribute funds, create their own  I proposal and fully fund it, there's no reason they couldn't do that. I don't see how having a fungible token  adds anything to this at this point that justifies the complexity of adding the full ICO functionality into this POC.

**Reply by falkonprods** [reply]
Dont' forget: not every member is a voting member. Not everyone gets to decide what happens. Which means that the DAO gets to decide who can allocate tokens.

**Reply by Zk2u** [reply]
> Dont' forget: not every member is a voting member. Not everyone gets to decide what happens. Which means that the DAO gets to decide who can allocate tokens.

This should be on a DAO-by-DAO choice though. For example, in @syvita, everyone will be a voting member and it will follow a similar model as @JMKJR says in this RFC (I think). But in Freehold, not everyone is. This should be configurable as some sort of plugin, where that be a new internal component inside the DAO contract, or an external plugin. 

Personally, it makes sense for me for this to be in the DAO contract with other essential components such as the treasury engine. But that's for another discussion...

**Reply by JMKJR** [reply]
> I am not sure exactly what the question is but maybe walking through a couple scenarios might help:

Thank you for these scenarios. They help frame the discussion and I better understand how we're seeing things differently. I believe the different governance model solutions we are proposing are two different solutions for entirely different users/stakeholders. I come from pre-conceived notions of a DAO as internet strangers who get together to create products, manage a treasury, vote on product initiatives etc. i.e they bootstrap a fair governance token initiative and try to create something from nothing in an entrepreneurial fashion with tokenomics permitting a trustless environment with high motivation and proclivity for action. This existing view of mine comes from my experiences observing the success of the Ethereum network and recently successful DAO's there. I see this as the biggest empirical DAO use case and one that could ultimately service us (Syvita) as a team building out the metaverse. I never pictured a DAO as having "assigned voting members by management" like Freehold is asking for. I don't see how this governance model scales well to a group of internet anons.

Scenario 1:
> So they vote on this, and eventually they make it to the 100 STX that was bid to build this out.

So basically we have an objective and any approved voting member can vote by staking funds to this objective? What's in it for the bidder/staker? This is back to my question about charity/good will. Why should voting member A stake a disproportionate amount of STX than voter B to fund the Clarity Video objective? Because they really want it done? I guess if this is a charitable foundation/organization that's a good enough reason.

Scenario 2:
Same questions as scenario 1.
> The purpose of a DAO is to come together to GET SOMETHING DONE

Yes! Which is why I think the profit incentive of token equity is the best catalyst for getting things done! Maybe you all disagree.

Scenario 3:
I see the treasury as separate from the issue of reclaiming "funds staked to a proposal". Multiple proposals could theoretically occur at once with an entirely separate DAO treasury holding unrelated funds, no?

> I don't see how having a fungible token adds anything to this at this point that justifies the complexity of adding the full ICO functionality into this POC.

I was never suggesting integrating ICO functionality.

Conclusion:
**I want to be clear that I do not feel Freehold or Stacks Foundation need a governance token per this RFC and I was never suggesting that**. I question whether they need to be a DAO at all. If they do, I see @falkonprods 's current governance model vision as good for a charitable organization or foundation that wants a better platform to fund objectives and distribute funds. This RFC is more so to service the stakeholders that I described at the beginning of this comment since I see that as the bigger use case for daoOS. If we cannot pivot due to the grant ramifications and the stakeholders we service and/or the Syvita team does not agree with this "bigger use case" conclusion, then let's just move ahead and build this out for Freehold and Stacks Foundation.

**Reply by falkonprods** [reply]
The purpose of daoOS should be to create a flexible model for forming DAOs.  I don't see how anything we are doing goes against that, and I don't see why we can't build a scalable foundation that can incorporate fungible tokens at a later date. There are myriad types of DAOs and governing methodologies. I don't think it makes sense to get overly hung up on this point at current. Honestly I think the current model is more of a proposal execution methodology and doesn't address all details of DAO governance. It  is an MVP,  a POC that was requested when I made my grant proposal, so I think we need to just get this built, and think about  what we can do with subsequent grants - and how we want to govern Syvita -  and use that dialogue to build this out further.

**Reply by 314159265359879** [reply]
I agree, build the MVP/ POC requested, make subsequent plans for other grant proposals to cover more governing methodologies DAO's may want to adopt.

**Reply by JMKJR** [reply]
I'm on board with the team's decision to build out the current vision for Stacks/Freehold. Let's move forward.

**Reply by louiseivan** [reply]
@falkonprods I think what @JMKJR is saying is more so the grandeur vision of daoOS, having a token-based project have a lot of upside to it as he described in the initial thread—A token-based project will accelerate growth and allows for faster innovation. By the end of the day, crypto is very fast-paced and success doesn't last that long. What you guys are doing now is an excellent proof of concept and can be taken to the next level such as BadgerDAO. 

I imagine a world where anyone can launch any project in daoOS using its native token. STX are used for Smart Contracts, and there's a great incentive to hold, making users spend is a different problem; Each token has its own unique cases. To your point on just using STX, it is saying the same as to why we shouldn't use BTC for daoOS. I'm a big believer in currency competition to foster innovation.

Also, don't want you guys to get sidetracked so +1 on focusing on the milestone but this is definitely something to put at the back of your mind.
